src/main/java/org/javarosa/core/util/CompressingIdGenerator.java

```diff
@@ -0,0 +1,118 @@
+package org.javarosa.core.util;
+
+/**
+ * Created by ctsims on 3/24/2017.
+ */
+
+public class CompressingIdGenerator {
+
+
+    /**
+     * Compresses the provided input number into a string.
+     *
+     * Requires three defined symbol spaces to transform the input string, Growth, Lead, and Body.
+     *
+     * The growth and lead symbol spaces should be mutually exclusive.
+     *
+     * Multiple strings generated using the same symbol spaces can be concatenated together in
+     * such a way that they will always be unique within their inputs
+     *
+     * The resulting ID will be of the form
+     *
+     * [G]*LBBBB
+     *
+     * Where G is a dynamic number of digits from the "Growth" symbol space, L is a single digit
+     * from the "Lead" symbol space, and B is a fixed number of digits from the "Body" symbol
+     * space.
+     *
+     * Note that there is *always* exactly one "Lead" digit. It is acceptable for the count of
+     * "Body" digits to be 0.
+     *
+     * @param input a number to be encoded by the scheme provided. Must be a positive integer
+     * @param bodyDigitCount the fixed number of "Body" digits that will be used to encode the
+     *                       input value
+     */
+    public static String generateCompressedIdString(long input,
```

Comment:
I'm not familiar with this encoding schema; curious whether this based on some established protocol, or just something you created?